### ISSUE TITLE
Add cache for rust build

### ISSUE THREAD

**Issue by sevenautumns** [DLR-FT/a653rs-linux#37]
If we want to improve compilation times more we could add a cache like #19 for the rust build as well.

**Comment by Steav005** [comment]
https://github.com/marketplace/actions/rust-cache

**Comment by wucke13** [comment]
How about something like this:

```yaml
jobs:
  checks:
    name: Nix Checks
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Cache Nix store
        uses: actions/cache@v3
        id: nix-cache
        with:
          path: /tmp/nixcache
          key: nix-${{ runner.os }}
      - uses: cachix/install-nix-action@v13
        with:
          nix_path: nixpkgs=channel:nixos-stable
          extra_nix_config: |
            experimental-features = nix-command flakes
            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
      - name: Import Nix store cache
        if: "steps.nix-cache.outputs.cache-hit == 'true'"
        run: nix copy --no-check-sigs --from /tmp/nixcache --all
      - name: Run Checks
        run: nix flake check . --print-build-logs
      - name: Export Nix store cache
        if: "steps.nix-cache.outputs.cache-hit != 'true'"
        run: |
          nix copy --no-check-sigs --to /tmp/nixcache \
            .#default \
            .#other-pkg
```

**Comment by wucke13** [comment]
Fixed by 0fdd1035
